Add ability to skip tests on specific helix queues

HaoK opened the pull request:
For https://github.com/aspnet/AspNetCore/issues/8169

HaoK committed:
Pass helix target queue to runtests

HaoK committed:
Use the helixQueue instead of true in the helix env

HaoK committed:
Set helix queue

HaoK committed:
Set helix queue to env

HaoK committed:
SkipOnHelix is when helix env is set to anything

HaoK committed:
Add queues to SkipOnHelix

HaoK committed:
See if target queue skipping works

Tratcher commented on src/Shared/test/SkipOnHelixAttribute.cs:
```diff
@@ -16,10 +15,15 @@ public bool IsMet
             {
-                return !OnHelix();
+                // Skip 
+                var skip = OnHelix() && (Queues == null || Queues.Contains(GetTargetHelixQueue(), StringComparison.OrdinalIgnoreCase));
```
This can get you odd partial matches. You should split on `;` and then do List.Contains.

HaoK committed:
More robust contains

HaoK committed:
Normalize strings to lower

HaoK committed:
Update SkipOnHelixAttribute.cs

HaoK commented:
Capturing target skip from https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190306.30/workItem/Microsoft.AspNetCore.Authentication.Test-netcoreapp3.0/wilogs before reverting the extra skips

``` Skipped on OSX
[xUnit.net 00:00:01.19]     Microsoft.AspNetCore.Authentication.DataHandler.SecureDataFormatTests.ProtectDataRoundTrips [SKIP]
Skipped  Microsoft.AspNetCore.Authentication.DataHandler.SecureDataFormatTests.ProtectDataRoundTrips
```

``` Skipped both on windows
Skipped  Microsoft.AspNetCore.Authentication.DataHandler.SecureDataFormatTests.ProtectWithPurposeRoundTrips
Skipped  Microsoft.AspNetCore.Authentication.DataHandler.SecureDataFormatTests.ProtectDataRoundTrips
```

HaoK committed:
Revert test skips

HaoK commented:
I'm rerunning the build, but @aspnet/brt signal r test failure on Redhat 7

https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190306.43/workItem/Microsoft.AspNetCore.SignalR.Client.FunctionalTests-netcoreapp3.0/wilogs
```
Starting test execution, please wait...
[xUnit.net 00:00:05.86]     Microsoft.AspNetCore.SignalR.Client.FunctionalTests.HubConnectionTests.ServerThrowsHubExceptionIfHubMethodCannotBeResolved(hubProtocolName: "json", transportType: LongPolling, hubPath: "/dynamic") [FAIL]
Failed   Microsoft.AspNetCore.SignalR.Client.FunctionalTests.HubConnectionTests.ServerThrowsHubExceptionIfHubMethodCannotBeResolved(hubProtocolName: "json", transportType: LongPolling, hubPath: "/dynamic")
Error Message:
 System.Exception : 3 error(s) logged.
Microsoft.AspNetCore.Http.Connections.Internal.Transports.LongPollingTransport - LongPollingTerminated - 6p83XxLNphaxJzMkEsEMOA - Long Polling transport was terminated due to an error on connection.
===================
System.InvalidOperationException: Reading is not allowed after reader was completed.
   at System.IO.Pipelines.ThrowHelper.ThrowInvalidOperationException_NoReadingAllowed()
   at System.IO.Pipelines.Pipe.ReadAsync(CancellationToken token)
   at System.IO.Pipelines.Pipe.DefaultPipeReader.ReadAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Http.Connections.Internal.Transports.LongPollingTransport.ProcessRequestAsync(HttpContext context, CancellationToken token) in /_/src/SignalR/common/Http.Connections/src/Internal/Transports/LongPollingTransport.cs:line 31
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher.ExecuteAsync(HttpContext context, ConnectionDelegate connectionDelegate, HttpConnectionDispatcherOptions options, ConnectionLogScope logScope) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:line 300
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionContext.WaitOnTasks(Task applicationTask, Task transportTask, Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs:line 262
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher.ExecuteAsync(HttpContext context, HttpConnectionDispatcherOptions options, ConnectionDelegate connectionDelegate) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:line 78
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context) in /_/src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs:line 57
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionContext.DisposeAsync(Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs:line 222
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext) in /_/src/Http/Routing/src/EndpointRoutingMiddleware.cs:line 78
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionManager.DisposeAndRemoveAsync(HttpConnectionContext connection, Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionManager.cs:line 205
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context) in /_/src/Security/Authentication/Core/src/AuthenticationMiddleware.cs:line 61
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application) in /_/src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs:line 576
===================
Microsoft.AspNetCore.Server.Kestrel - ApplicationError - Connection id "0HLL2IDNN89U9", Request id "0HLL2IDNN89U9:00000005": An unhandled exception was thrown by the application.
===================
System.InvalidOperationException: Reading is not allowed after reader was completed.
   at System.IO.Pipelines.ThrowHelper.ThrowInvalidOperationException_NoReadingAllowed()
   at System.IO.Pipelines.Pipe.ReadAsync(CancellationToken token)
   at System.IO.Pipelines.Pipe.DefaultPipeReader.ReadAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Http.Connections.Internal.Transports.LongPollingTransport.ProcessRequestAsync(HttpContext context, CancellationToken token) in /_/src/SignalR/common/Http.Connections/src/Internal/Transports/LongPollingTransport.cs:line 31
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher.ExecuteAsync(HttpContext context, ConnectionDelegate connectionDelegate, HttpConnectionDispatcherOptions options, ConnectionLogScope logScope) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:line 300
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionContext.WaitOnTasks(Task applicationTask, Task transportTask, Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs:line 262
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher.ExecuteAsync(HttpContext context, HttpConnectionDispatcherOptions options, ConnectionDelegate connectionDelegate) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:line 78
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context) in /_/src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs:line 57
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionContext.DisposeAsync(Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs:line 222
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext) in /_/src/Http/Routing/src/EndpointRoutingMiddleware.cs:line 78
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionManager.DisposeAndRemoveAsync(HttpConnectionContext connection, Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionManager.cs:line 205
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context) in /_/src/Security/Authentication/Core/src/AuthenticationMiddleware.cs:line 61
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application) in /_/src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs:line 576
===================
Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionManager - FailedDispose - Failed disposing connection 6p83XxLNphaxJzMkEsEMOA.
===================
System.InvalidOperationException: Reading is not allowed after reader was completed.
   at System.IO.Pipelines.ThrowHelper.ThrowInvalidOperationException_NoReadingAllowed()
   at System.IO.Pipelines.Pipe.ReadAsync(CancellationToken token)
   at System.IO.Pipelines.Pipe.DefaultPipeReader.ReadAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Http.Connections.Internal.Transports.LongPollingTransport.ProcessRequestAsync(HttpContext context, CancellationToken token) in /_/src/SignalR/common/Http.Connections/src/Internal/Transports/LongPollingTransport.cs:line 31
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher.ExecuteAsync(HttpContext context, ConnectionDelegate connectionDelegate, HttpConnectionDispatcherOptions options, ConnectionLogScope logScope) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:line 300
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionContext.WaitOnTasks(Task applicationTask, Task transportTask, Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs:line 262
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher.ExecuteAsync(HttpContext context, HttpConnectionDispatcherOptions options, ConnectionDelegate connectionDelegate) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:line 78
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context) in /_/src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs:line 57
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionContext.DisposeAsync(Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs:line 222
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext) in /_/src/Http/Routing/src/EndpointRoutingMiddleware.cs:line 78
   at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionManager.DisposeAndRemoveAsync(HttpConnectionContext connection, Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionManager.cs:line 205
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context) in /_/src/Security/Authentication/Core/src/AuthenticationMiddleware.cs:line 61
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application) in /_/src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs:line 576
===================
Stack Trace:
   at Microsoft.AspNetCore.SignalR.Tests.VerifyNoErrorsScope.Dispose() in E:\A\_work\3\s\src\SignalR\common\testassets\Tests.Utils\VerifyNoErrorsScope.cs:line 63
   at Microsoft.AspNetCore.SignalR.Tests.FunctionalTestBase.MultiDisposable.Dispose() in E:\A\_work\3\s\src\SignalR\common\testassets\Tests.Utils\FunctionalTestBase.cs:line 55
   at Microsoft.AspNetCore.SignalR.Client.FunctionalTests.HubConnectionTests.ServerThrowsHubExceptionIfHubMethodCannotBeResolved(String hubProtocolName, HttpTransportType transportType, String hubPath) in /_/src/SignalR/clients/csharp/Client/test/FunctionalTests/HubConnectionTests.cs:line 536
--- End of stack trace from previous location where exception was thrown ---
Standard Output Messages:
 | [0.003s] TestLifetime Information: Starting test ServerThrowsHubExceptionIfHubMethodCannotBeResolved-json-LongPolling-/dynamic at 2019-03-07T00:25:45
 | [0.013s] Microsoft.AspNetCore.SignalR.Tests.InProcessTestServer Information: Starting test server...
 | [0.014s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting starting
 | [0.017s] Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager Information: User profile is available. Using '/home/helixbot/.aspnet/DataProtection-Keys' as key repository; keys will not be encrypted at rest.
 | [0.018s] Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository Debug: Reading data from file '/home/helixbot/.aspnet/DataProtection-Keys/key-379a24c1-6530-448d-85f9-a1b213bca234.xml'.
 | [0.022s] Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager Debug: Found key {379a24c1-6530-448d-85f9-a1b213bca234}.
 | [0.023s] Microsoft.AspNetCore.DataProtection.KeyManagement.DefaultKeyResolver Debug: Considering key {379a24c1-6530-448d-85f9-a1b213bca234} with expiration date 2019-06-05 00:25:43Z as default key.
 | [0.024s] Microsoft.AspNetCore.DataProtection.TypeForwardingActivator Debug: Forwarded activator type request from Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.AuthenticatedEncryptorDescriptorDeserializer, Microsoft.AspNetCore.DataProtection, Version=3.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 to Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.AuthenticatedEncryptorDescriptorDeserializer, Microsoft.AspNetCore.DataProtection, Culture=neutral, PublicKeyToken=adb9793829ddae60
 | [0.025s] Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ManagedAuthenticatedEncryptorFactory Debug: Using managed symmetric algorithm 'System.Security.Cryptography.Aes'.
 | [0.025s] Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ManagedAuthenticatedEncryptorFactory Debug: Using managed keyed hash algorithm 'System.Security.Cryptography.HMACSHA256'.
 | [0.026s] Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingProvider Debug: Using key {379a24c1-6530-448d-85f9-a1b213bca234} as the default key.
 | [0.026s] Microsoft.AspNetCore.DataProtection.Internal.DataProtectionStartupFilter Debug: Key ring with default key {379a24c1-6530-448d-85f9-a1b213bca234} was loaded during application startup.
 | [0.029s] Microsoft.AspNetCore.SignalR.Internal.DefaultHubProtocolResolver Debug: Registered SignalR Protocol: json, implemented by Microsoft.AspNetCore.SignalR.Protocol.NewtonsoftJsonHubProtocol.
 | [0.029s] Microsoft.AspNetCore.SignalR.Internal.DefaultHubProtocolResolver Debug: Registered SignalR Protocol: messagepack, implemented by Microsoft.AspNetCore.SignalR.Protocol.MessagePackHubProtocol.
 | [0.090s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting started
 | [0.091s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Loaded hosting startup assembly Microsoft.AspNetCore.SignalR.Client.FunctionalTests
 | [0.091s] Microsoft.AspNetCore.SignalR.Tests.InProcessTestServer Information: Test Server started
 | [0.092s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Starting HubConnection.
 | [0.093s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Starting HttpConnection.
 | [0.093s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Establishing connection with server at 'http://127.0.0.1:41196/dynamic'.
 | [0.098s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89U9" started.
 | [0.100s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request starting HTTP/1.1 POST http://127.0.0.1:41196/dynamic/negotiate  0
 | [0.101s] Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler Debug: AuthenticationScheme: Bearer was not authenticated.
 | [0.105s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: 1 candidate(s) found for the request path '/dynamic/negotiate'
 | [0.105s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: Endpoint '/dynamic/negotiate' with route pattern '/dynamic/negotiate' is valid for the request path '/dynamic/negotiate'
 | [0.105s] Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware Debug: Request matched endpoint '/dynamic/negotiate'
 | [0.106s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionManager Debug: New connection 6p83XxLNphaxJzMkEsEMOA created.
 | [0.106s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher Debug: 6p83XxLNphaxJzMkEsEMOA - Sending negotiation response.
 | [0.107s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89U9" completed keep alive response.
 | [0.107s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request finished in 6.6795ms 200 application/json
 | [0.115s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Established connection '6p83XxLNphaxJzMkEsEMOA' with the server.
 | [0.116s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Skipping transport WebSockets because it was disabled by the client.
 | [0.116s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Skipping transport ServerSentEvents because it was disabled by the client.
 | [0.117s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Starting transport 'LongPolling' with Url: http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA.
 | [0.117s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Information: Starting transport. Transfer mode: Text.
 | [0.119s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request starting HTTP/1.1 GET http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA  
 | [0.120s] Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler Debug: AuthenticationScheme: Bearer was not authenticated.
 | [0.123s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: 1 candidate(s) found for the request path '/dynamic'
 | [0.123s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: Endpoint '/dynamic' with route pattern '/dynamic' is valid for the request path '/dynamic'
 | [0.123s] Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware Debug: Request matched endpoint '/dynamic'
 | [0.124s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher Debug: 6p83XxLNphaxJzMkEsEMOA - Establishing new connection.
 | [0.125s] Microsoft.AspNetCore.SignalR.HubConnectionHandler Debug: 6p83XxLNphaxJzMkEsEMOA - OnConnectedAsync started.
 | [0.126s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89U9" completed keep alive response.
 | [0.127s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request finished in 8.5238ms 200 application/octet-stream
 | [0.128s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Starting receive loop.
 | [0.129s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Starting the send loop.
 | [0.129s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Transport 'LongPolling' started.
 | [0.130s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request starting HTTP/1.1 GET http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA  
 | [0.132s] Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler Debug: AuthenticationScheme: Bearer was not authenticated.
 | [0.132s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: 1 candidate(s) found for the request path '/dynamic'
 | [0.132s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: Endpoint '/dynamic' with route pattern '/dynamic' is valid for the request path '/dynamic'
 | [0.132s] Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware Debug: Request matched endpoint '/dynamic'
 | [0.133s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher Debug: 6p83XxLNphaxJzMkEsEMOA - Resuming existing connection.
 | [0.134s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Information: HttpConnection Started.
 | [0.134s] Microsoft.AspNetCore.SignalR.Client.HubConnection Information: Using HubProtocol 'json v1'.
 | [0.135s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Sending Hub Handshake.
 | [0.141s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Sending 32 bytes to the server using url: http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA.
 | [0.143s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89UA" started.
 | [0.149s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request starting HTTP/1.1 POST http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA  32
 | [0.150s] Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler Debug: AuthenticationScheme: Bearer was not authenticated.
 | [0.150s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: 1 candidate(s) found for the request path '/dynamic'
 | [0.150s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: Endpoint '/dynamic' with route pattern '/dynamic' is valid for the request path '/dynamic'
 | [0.150s] Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware Debug: Request matched endpoint '/dynamic'
 | [0.152s] Microsoft.AspNetCore.Server.Kestrel Debug: 6p83XxLNphaxJzMkEsEMOA - Connection id "0HLL2IDNN89UA", Request id "0HLL2IDNN89UA:00000001": started reading request body.
 | [0.152s] Microsoft.AspNetCore.Server.Kestrel Debug: 6p83XxLNphaxJzMkEsEMOA - Connection id "0HLL2IDNN89UA", Request id "0HLL2IDNN89UA:00000001": done reading request body.
 | [0.152s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89UA" completed keep alive response.
 | [0.153s] Microsoft.AspNetCore.SignalR.Internal.DefaultHubProtocolResolver Debug: 6p83XxLNphaxJzMkEsEMOA - Found protocol implementation for requested protocol: json.
 | [0.155s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request finished in 5.8673ms 200 text/plain
 | [0.155s] Microsoft.AspNetCore.SignalR.HubConnectionContext Information: 6p83XxLNphaxJzMkEsEMOA - Completed connection handshake. Using HubProtocol 'json'.
 | [0.156s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Message(s) sent successfully.
 | [0.157s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89U9" completed keep alive response.
 | [0.157s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request finished in 27.7682ms 200 application/octet-stream
 | [0.160s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Received messages from the server.
 | [0.160s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Handshake with server complete.
 | [0.160s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Receive loop starting.
 | [0.161s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Sending PingMessage message.
 | [0.161s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Sending PingMessage message completed.
 | [0.161s] Microsoft.AspNetCore.SignalR.Client.HubConnection Information: HubConnection started.
 | [0.162s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Sending 11 bytes to the server using url: http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA.
 | [0.163s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request starting HTTP/1.1 GET http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA  
 | [0.164s] Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler Debug: AuthenticationScheme: Bearer was not authenticated.
 | [0.164s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: 1 candidate(s) found for the request path '/dynamic'
 | [0.164s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: Endpoint '/dynamic' with route pattern '/dynamic' is valid for the request path '/dynamic'
 | [0.164s] Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware Debug: Request matched endpoint '/dynamic'
 | [0.164s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher Debug: 6p83XxLNphaxJzMkEsEMOA - Resuming existing connection.
 | [0.165s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Registering Invocation ID '1' for tracking.
 | [0.166s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Sending InvocationMessage message '1'.
 | [0.166s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Sending InvocationMessage message '1' completed.
 | [0.166s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request starting HTTP/1.1 POST http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA  11
 | [0.167s] Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler Debug: AuthenticationScheme: Bearer was not authenticated.
 | [0.167s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: 1 candidate(s) found for the request path '/dynamic'
 | [0.167s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: Endpoint '/dynamic' with route pattern '/dynamic' is valid for the request path '/dynamic'
 | [0.167s] Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware Debug: Request matched endpoint '/dynamic'
 | [0.167s] Microsoft.AspNetCore.Server.Kestrel Debug: 6p83XxLNphaxJzMkEsEMOA - Connection id "0HLL2IDNN89UA", Request id "0HLL2IDNN89UA:00000002": started reading request body.
 | [0.167s] Microsoft.AspNetCore.Server.Kestrel Debug: 6p83XxLNphaxJzMkEsEMOA - Connection id "0HLL2IDNN89UA", Request id "0HLL2IDNN89UA:00000002": done reading request body.
 | [0.168s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89UA" completed keep alive response.
 | [0.168s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request finished in 1.3937000000000002ms 200 text/plain
 | [0.169s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Message(s) sent successfully.
 | [0.169s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Sending 62 bytes to the server using url: http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA.
 | [0.170s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request starting HTTP/1.1 POST http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA  62
 | [0.171s] Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler Debug: AuthenticationScheme: Bearer was not authenticated.
 | [0.171s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: 1 candidate(s) found for the request path '/dynamic'
 | [0.171s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: Endpoint '/dynamic' with route pattern '/dynamic' is valid for the request path '/dynamic'
 | [0.171s] Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware Debug: Request matched endpoint '/dynamic'
 | [0.171s] Microsoft.AspNetCore.Server.Kestrel Debug: 6p83XxLNphaxJzMkEsEMOA - Connection id "0HLL2IDNN89UA", Request id "0HLL2IDNN89UA:00000003": started reading request body.
 | [0.171s] Microsoft.AspNetCore.Server.Kestrel Debug: 6p83XxLNphaxJzMkEsEMOA - Connection id "0HLL2IDNN89UA", Request id "0HLL2IDNN89UA:00000003": done reading request body.
 | [0.172s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89UA" completed keep alive response.
 | [0.172s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request finished in 1.4119000000000002ms 200 text/plain
 | [0.172s] Microsoft.AspNetCore.SignalR.Internal.DefaultHubDispatcher Error: 6p83XxLNphaxJzMkEsEMOA - Failed to invoke hub method '!@#$%'.
 | Microsoft.AspNetCore.SignalR.HubException: Method does not exist.
 |    at Microsoft.AspNetCore.SignalR.Internal.DefaultHubDispatcher`1.GetParameterTypes(String methodName) in /_/src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs:line 566
 |    at Microsoft.AspNetCore.SignalR.Internal.HubConnectionBinder`1.GetParameterTypes(String methodName) in /_/src/SignalR/server/Core/src/Internal/HubConnectionBinder.cs:line 23
 |    at Microsoft.AspNetCore.SignalR.Protocol.NewtonsoftJsonHubProtocol.ParseMessage(Utf8BufferTextReader textReader, IInvocationBinder binder) in /_/src/SignalR/common/Protocols.NewtonsoftJson/src/Protocol/NewtonsoftJsonHubProtocol.cs:line 265
 | [0.174s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Message(s) sent successfully.
 | [0.174s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Received messages from the server.
 | [0.175s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Processing 133 byte message from server.
 | [0.176s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Stopping HubConnection.
 | [0.176s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Terminating receive loop.
 | [0.176s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Waiting for the receive loop to terminate.
 | [0.177s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Disposing HttpConnection.
 | [0.177s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Information: Transport is stopping.
 | [0.177s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Send loop canceled.
 | [0.177s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Send loop stopped.
 | [0.179s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Receive loop stopped.
 | [0.179s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Sending DELETE request to 'http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA'.
 | [0.179s] Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets Debug: Connection id "0HLL2IDNN89U9" received FIN.
 | [0.180s] Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets Debug: Connection id "0HLL2IDNN89U9" sending FIN because: "The client closed the connection."
 | [0.181s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89U9" disconnecting.
 | [0.184s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request starting HTTP/1.1 DELETE http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA  
 | [0.184s] Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler Debug: AuthenticationScheme: Bearer was not authenticated.
 | [0.185s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: 1 candidate(s) found for the request path '/dynamic'
 | [0.185s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: Endpoint '/dynamic' with route pattern '/dynamic' is valid for the request path '/dynamic'
 | [0.185s] Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware Debug: Request matched endpoint '/dynamic'
 | [0.186s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89UA" completed keep alive response.
 | [0.186s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request finished in 1.6509ms 202 text/plain
 | [0.187s] Microsoft.AspNetCore.SignalR.HubConnectionHandler Debug: 6p83XxLNphaxJzMkEsEMOA - OnConnectedAsync ending.
 | [0.187s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request finished in 23.8975ms 200 application/octet-stream
 | [0.189s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request starting HTTP/1.1 GET http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA  
 | [0.191s] Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler Debug: AuthenticationScheme: Bearer was not authenticated.
 | [0.192s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: 1 candidate(s) found for the request path '/dynamic'
 | [0.192s] Microsoft.AspNetCore.Routing.Matching.DfaMatcher Debug: Endpoint '/dynamic' with route pattern '/dynamic' is valid for the request path '/dynamic'
 | [0.192s] Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware Debug: Request matched endpoint '/dynamic'
 | [0.195s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher Debug: 6p83XxLNphaxJzMkEsEMOA - Resuming existing connection.
 | [0.198s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: DELETE request to 'http://127.0.0.1:41196/dynamic?id=6p83XxLNphaxJzMkEsEMOA' accepted.
 | [0.199s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Transport stopped.
 | [0.199s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Information: HttpConnection Disposed.
 | [0.199s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Canceling all outstanding invocations.
 | [0.199s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: HubConnection stopped.
 | [0.199s] Microsoft.AspNetCore.SignalR.Tests.InProcessTestServer Information: Shutting down test server
 | [0.200s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting shutdown
 | [0.200s] Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets Debug: Connection id "0HLL2IDNN89UA" received FIN.
 | [0.200s] Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets Debug: Connection id "0HLL2IDNN89UA" sending FIN because: "The client closed the connection."
 | [0.200s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89UA" disconnecting.
 | [0.201s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89UA" stopped.
 | [0.198s] Microsoft.AspNetCore.Http.Connections.Internal.Transports.LongPollingTransport Error: 6p83XxLNphaxJzMkEsEMOA - Long Polling transport was terminated due to an error on connection.
 | System.InvalidOperationException: Reading is not allowed after reader was completed.
 |    at System.IO.Pipelines.ThrowHelper.ThrowInvalidOperationException_NoReadingAllowed()
 |    at System.IO.Pipelines.Pipe.ReadAsync(CancellationToken token)
 |    at System.IO.Pipelines.Pipe.DefaultPipeReader.ReadAsync(CancellationToken cancellationToken)
 |    at Microsoft.AspNetCore.Http.Connections.Internal.Transports.LongPollingTransport.ProcessRequestAsync(HttpContext context, CancellationToken token) in /_/src/SignalR/common/Http.Connections/src/Internal/Transports/LongPollingTransport.cs:line 31
 | [0.203s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionManager Error: Failed disposing connection 6p83XxLNphaxJzMkEsEMOA.
 | System.InvalidOperationException: Reading is not allowed after reader was completed.
 |    at System.IO.Pipelines.ThrowHelper.ThrowInvalidOperationException_NoReadingAllowed()
 |    at System.IO.Pipelines.Pipe.ReadAsync(CancellationToken token)
 |    at System.IO.Pipelines.Pipe.DefaultPipeReader.ReadAsync(CancellationToken cancellationToken)
 |    at Microsoft.AspNetCore.Http.Connections.Internal.Transports.LongPollingTransport.ProcessRequestAsync(HttpContext context, CancellationToken token) in /_/src/SignalR/common/Http.Connections/src/Internal/Transports/LongPollingTransport.cs:line 31
 |    at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher.ExecuteAsync(HttpContext context, ConnectionDelegate connectionDelegate, HttpConnectionDispatcherOptions options, ConnectionLogScope logScope) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:line 300
 |    at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionContext.WaitOnTasks(Task applicationTask, Task transportTask, Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs:line 262
 |    at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher.ExecuteAsync(HttpContext context, HttpConnectionDispatcherOptions options, ConnectionDelegate connectionDelegate) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:line 78
 |    at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context) in /_/src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs:line 57
 |    at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionContext.DisposeAsync(Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs:line 222
 |    at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext) in /_/src/Http/Routing/src/EndpointRoutingMiddleware.cs:line 78
 |    at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionManager.DisposeAndRemoveAsync(HttpConnectionContext connection, Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionManager.cs:line 205
 |    at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context) in /_/src/Security/Authentication/Core/src/AuthenticationMiddleware.cs:line 61
 |    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application) in /_/src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs:line 576
 | [0.203s] Microsoft.AspNetCore.Server.Kestrel Error: Connection id "0HLL2IDNN89U9", Request id "0HLL2IDNN89U9:00000005": An unhandled exception was thrown by the application.
 | System.InvalidOperationException: Reading is not allowed after reader was completed.
 |    at System.IO.Pipelines.ThrowHelper.ThrowInvalidOperationException_NoReadingAllowed()
 |    at System.IO.Pipelines.Pipe.ReadAsync(CancellationToken token)
 |    at System.IO.Pipelines.Pipe.DefaultPipeReader.ReadAsync(CancellationToken cancellationToken)
 |    at Microsoft.AspNetCore.Http.Connections.Internal.Transports.LongPollingTransport.ProcessRequestAsync(HttpContext context, CancellationToken token) in /_/src/SignalR/common/Http.Connections/src/Internal/Transports/LongPollingTransport.cs:line 31
 |    at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher.ExecuteAsync(HttpContext context, ConnectionDelegate connectionDelegate, HttpConnectionDispatcherOptions options, ConnectionLogScope logScope) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:line 300
 |    at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionContext.WaitOnTasks(Task applicationTask, Task transportTask, Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs:line 262
 |    at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher.ExecuteAsync(HttpContext context, HttpConnectionDispatcherOptions options, ConnectionDelegate connectionDelegate) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:line 78
 |    at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context) in /_/src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs:line 57
 |    at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionContext.DisposeAsync(Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs:line 222
 |    at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext) in /_/src/Http/Routing/src/EndpointRoutingMiddleware.cs:line 78
 |    at Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionManager.DisposeAndRemoveAsync(HttpConnectionContext connection, Boolean closeGracefully) in /_/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionManager.cs:line 205
 |    at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context) in /_/src/Security/Authentication/Core/src/AuthenticationMiddleware.cs:line 61
 |    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application) in /_/src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs:line 576
 | [0.206s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionManager Debug: Removing connection 6p83XxLNphaxJzMkEsEMOA from the list of connections.
 | [0.207s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request finished in 17.7153ms 0 text/plain
 | [0.207s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLL2IDNN89U9" stopped.
 | [0.207s] Microsoft.AspNetCore.SignalR.Tests.InProcessTestServer Information: Test server shut down
 | [0.212s] TestLifetime Information: Finished test ServerThrowsHubExceptionIfHubMethodCannotBeResolved-json-LongPolling-/dynamic in 0.20900159999999998s


Results File: /home/helixbot/dotnetbuild/work/d026df67-c7e6-4c3f-936a-f57c3b628992/Work/c0f65088-2486-471c-be70-e1114cf297c5/Exec/TestResults/_a00138T_2019-03-07_00_25_44_043.trx

Total tests: 363. Passed: 362. Failed: 1. Skipped: 0.
```

dougbu reviewed (approved):
Looks good though Helix tests aren't done. Main question is overlap with `[FlakyTest]`. Is this needed?

HaoK commented:
Yeah, because we might need to skip tests on particular queue always, not just put it in jail.

anurse commented:
cc @BrennanConroy for SignalR failure. It's the Long Polling thing he and @halter73 are working on.
